src/posix/net.cpp

```diff
@@ -2357,6 +2432,29 @@ int CurlHttpIO::socket_callback(CURL *, curl_socket_t s, int what, void *userp,
     return 0;
 }
 
+int CurlHttpIO::sockopt_callback(void *clientp, curl_socket_t, curlsocktype)
```

Comment:
```this callback function gets called by libcurl when the socket has been created, but before the connect call to allow applications to change specific socket options```
We are already triggering a `gethostbyname()` when we attempt to `post()` a HTTP request in case there's no cached IP. Why do we need to control it here again? Is it to perform a lookup with every reconnection? (in case we are using a cached IP and not performing a lookup already) If so, no objections at all :)

Reply:
Maybe, if the above is right, we may want to add those conditions (`!httpio->disconnecting && !httpctx->ares_pending`) to prevent concurrent lookups when they are started at `post()`

Reply:
Yep. The specs say that a DNS request should be made in parallel with each new connection, so that's the intention.
`httpctx->isCachedIp` (few lines below) is intended to not send a new DNS request if it was already sent at `post()`.
`!httpio->disconnecting && !httpctx->ares_pending` are also in the check few lines below this comment.